autogpt_platform/backend/backend/blocks/airtable/_webhook.py

```diff
@@ -0,0 +1,161 @@
+"""
+Webhook management for Airtable blocks.
+"""
+
+import hashlib
+import hmac
+import logging
+from enum import Enum
+from typing import Tuple
+
+from backend.sdk import (
+    APIKeyCredentials,
+    BaseWebhooksManager,
+    Credentials,
+    ProviderName,
+    Requests,
+    Webhook,
+    update_webhook,
+)
+
+from ._api import (
+    WebhookFilters,
+    WebhookSpecification,
+    create_webhook,
+    list_webhook_payloads,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class AirtableWebhookManager(BaseWebhooksManager):
+    """Webhook manager for Airtable API."""
+
+    PROVIDER_NAME = ProviderName("airtable")
+
+    class WebhookType(str, Enum):
+        TABLE_CHANGE = "table_change"
+
+    @classmethod
+    async def validate_payload(
+        cls, webhook: Webhook, request, credentials: Credentials | None
+    ) -> Tuple[dict, str]:
+        """Validate incoming webhook payload and signature."""
+        payload = await request.json()
+
+        # Verify webhook signature using HMAC-SHA256
+        if webhook.secret:
+            mac_secret = webhook.config.get("mac_secret")
+            if mac_secret:
+                # Get the raw body for signature verification
+                body = await request.body()
+
+                # Calculate expected signature
+                mac_secret_decoded = mac_secret.encode()
+                hmac_obj = hmac.new(mac_secret_decoded, body, hashlib.sha256)
+                expected_mac = f"hmac-sha256={hmac_obj.hexdigest()}"
+
+                # Get signature from headers
+                signature = request.headers.get("X-Airtable-Content-MAC")
+
+                if signature and not hmac.compare_digest(signature, expected_mac):
+                    raise ValueError("Invalid webhook signature")
+
+        # Validate payload structure
+        required_fields = ["base", "webhook", "timestamp"]
+        if not all(field in payload for field in required_fields):
+            raise ValueError("Invalid webhook payload structure")
+
+        if "id" not in payload["base"] or "id" not in payload["webhook"]:
+            raise ValueError("Missing required IDs in webhook payload")
+
+        # get payload request parameters
+        base_id = webhook.config.get("base_id")
+        cursor = webhook.config.get("cursor")
+        webhook_id = webhook.config.get("webhook_id")
+
+        assert base_id is not None
+        assert webhook_id is not None
+        assert credentials is not None
```

Comment:
I'd advise checking this and handling the case of credentials not being available

Reply:
changed to raising errors - it won't work without credentials so we need to raise an error of some sort